adoc/deployment-aws.adoc

```diff
@@ -0,0 +1,87 @@
+== Deployment on Amazon AWS
+
+.Preparation Required
+[NOTE]
+You must have completed <<deployment.preparations>> to proceed.
+
+You will use {tf} to deploy the required master and worker cluster nodes (plus a load
+balancer) to {soc} and then use the `skuba` tool to bootstrap the {kube} cluster on
+top of those.
+
+. Download the AWS credentials
+.. Log in to the AWS console.
+.. Click on your username in the upper right hand corner to reveal the dropdown menu.
+.. Click on My Security Credentials.
+.. Click Create Access Key on the Security Credentials tab.
+.. Note down the newly created _Access_ and _Secret_ keys.
+
+=== Deploying the cluster nodes
+
+. Find the {tf} template files for {soc} in `/usr/share/caasp/terraform/aws`.
+Copy this folder to a location of your choice as the files need adjustment.
++
+----
+mkdir -p ~/caasp/deployment/
+cp -r /usr/share/caasp/terraform/aws/ ~/caasp/deployment/
+cd ~/caasp/deployment/aws/
+----
+. Once the files are copied, rename the `terraform.tfvars.example` file to
+`terraform.tfvars`:
++
+----
+mv terraform.tfvars.example terraform.tfvars
+----
+. Edit the `terraform.tfvars` file and add modify the following variables:
++
+include::deployment-terraform-example.adoc[tags=tf_aws]
++
+[TIP]
+====
+You can set timezone in before deploying the nodes by modifying the files:
+
+* `~/my-deployment/cloud-init/cloud-init.yaml.tpl`
+====
+. Now you can deploy the nodes by running:
++
+----
+terraform init
+terraform plan
+terraform apply
+----
++
+Check the output for the actions to be taken. Type "yes" and confirm with Enter when ready.
+Terraform will now provision all the machines and network infrastructure for the cluster.
+
+
+.Public IPs for nodes
+[IMPORTANT]
+====
+`skuba` cannot currently access nodes through a bastion host, so all
+the nodes in the cluster must be directly reachable from the machine where
+`skuba` is being run. We must also consider that `skuba` must use
+the external IPs as `--target`s when initializing or joining the cluster,
+while we must specify the internal DNS names for registering the nodes
+in the cluster.
+====
+
+.Note down IP/FQDN for nodes
+[IMPORTANT]
+====
+The IP addresses of the generated machines will be displayed in the terraform
+output during the cluster node deployment. You need these IP addresses to
+deploy {productname} to the cluster.
+
+If you need to find an IP addresses later on you can run `terraform output` within
+the directory you performed the deployment from `~/my-deployment` directory or
+perform the following steps:
+
+. Log in to {soc} and click on menu:Load Balancing[Load Balancers]. Find the one with the
```

Comment:
Again reference to `{soc}`. How do you deploy a load balancer for AWS?

Reply:
Fixed. The Load Balancer will be created ajutomatically from the Terraform code.